dowhy/causal_refuters/evalue_sensitivity_analyzer.py

```diff
@@ -0,0 +1,301 @@
+            )
+        if isinstance(
+            self.estimate.estimator,
+            (
+                Causalml,
+                Econml,
+                InstrumentalVariableEstimator,
+                RegressionDiscontinuityEstimator,
+                TwoStageRegressionEstimator,
+            ),
+        ):
+            raise NotImplementedError(
+                "E-Value sensitivity analysis is not currently implemented for this estimator"
+            )
+
+    def check_sensitivity(self):
+        """
+        Function to compute E-value for point estimate and confidence limits. The estimate and
+        confidence limits are converted to the risk ratio scale before the E-value is calculated.
+        """
+
+        est = self.estimate.value
+        se = self.estimate.get_standard_error()
+        if isinstance(se, np.ndarray):
+            se = se[0]
+
+        if isinstance(self.estimate.estimator, LinearRegressionEstimator):
+            self._evalue_OLS(est, se, self.sd_outcome, self.delta, self.true)
+        else:
+            est = self._to_md(est, self.sd_outcome)
+            se = self._to_md(se, self.sd_outcome)
+            self._evalue_MD(est, se, self.true)
+
+    def _evalue_OLS(self, est, se, sd, delta=1, true=0):
+        """
+        Function to compute E-value from OLS coefficient.
+
+        :param est: coefficient from OLS
+        :param se: standard error of point estimate
+        :param sd: residual standard deviation
+        :param delta: contrast of interest in the treatment/exposure
+        :param true: true standardized difference to which to shift the observed estimate. (Default = 0)
+        """
+        if est < 0:
+            raise ValueError("Standard error cannot be negative")
+
+        if delta < 0:
+            delta = -delta
+            self.logger.info("Recoding delta to be positive")
+
+        est = self._to_md(est, sd, delta=delta)
+        se = self._to_md(se, sd, delta=delta)
+        print(f"est {est}, se {se}")
+        self._evalue_MD(est=est, se=se, true=true)
+
+    def _evalue_MD(self, est, se=None, true=0):
+        """
+        Function to compute E-value for standardized difference and its confidence limits.
+
+        :param est: point estimate as standardized difference (i.e., Cohen's d)
+        :param se: standard error of the point estimate
+        :param true: true standardized difference to which to shift the observed estimate. (Default = 0)
+        """
+        if est < 0:
+            raise ValueError("Standard error cannot be negative")
+
+        if se is None:
+            lo = None
+            hi = None
+        else:
+            lo = np.exp(0.91 * est - 1.78 * se)
+            hi = np.exp(0.91 * est + 1.78 * se)
+
+        est = self._md_to_rr(est)
+        true = self._md_to_rr(true)
+
+        self._evalue_RR(est=est, lo=lo, hi=hi, true=true)
+
+    def _evalue_OR(self, est, lo=None, hi=None, rare=None, true=1):
```

Comment:
evalue OR and HR methods are not used. Which estimators will these be useful for?

Reply:
evalue OR can be to support`GeneralizedLinearModelEstimator` logistic regression. But it's probably safe to drop the hazard ratio method.

Reply:
Got it, makes sense.